### DISCUSSION TITLE
Selfhosted webserver

### DISCUSSION THREAD

**Discussion by raecer** [webrcade/webrcade#78] [Q&A] [answered]
How do you go about self-hosting the webserver and customizing the url and such? I can't find any docs on this or if it is available. Thank you.

**Comment by raz0red** [comment, marked as answer]
Hi-

Unfortunately it isn't documented (yet). There are a couple of stories (issues) to address this in an upcoming release:
https://github.com/webrcade/webrcade/issues/7#issuecomment-1230705867

But, there are a couple of ways to do it today:

1.) Self host the distribution package. Just serve this via any HTTP Server:

https://github.com/webrcade/webrcade/releases

Use one of the "-dist" packages for a recent release. The downside to this approach is that it doesn't support CORS, so any content that doesn't support cross origin requests will fail.

2.) Use the Docker image:

This is the preferred method as it has a CORS proxy built in.
https://hub.docker.com/r/webrcade/webrcade/

Here are some discussions related to Docker use:
https://github.com/webrcade/webrcade/issues/58#issuecomment-1207234171
https://github.com/webrcade/webrcade/issues/60#issuecomment-1230469838
